Python 3 support

janwijbrand opened the pull request.

janwijbrand committed:
first attempt at getting it to work with AST

janwijbrand committed:
from future import print_function

janwijbrand committed:
This false positive is gone.

janwijbrand committed:
Cleanups for when we support Python 3

janwijbrand committed:
remove commented prints

janwijbrand committed:
visit childs not the node itself over and over

janwijbrand committed:
update readme

janwijbrand committed:
manifest.in

janwijbrand committed:
rename

janwijbrand committed:
Preparing release 2.0+md.1

janwijbrand committed:
Back to development: 2.0+md.2

janwijbrand committed:
Merge remote-tracking branch 'origin/master' into OPS-714-python3

janwijbrand committed:
Update with master and prep for a proper PR to the zopefoundation repo

janwijbrand committed:
Merge remote-tracking branch 'origin/master' into OPS-714-python3

janwijbrand committed:
Merge pull request #2 from minddistrict/OPS-714-python3

Ops 714 python3

janwijbrand committed:
update version

janwijbrand committed:
Merge branch 'master' into master

janwijbrand committed:
correct python versions for travis runs

janwijbrand committed:
Merge branch 'master' of github.com:minddistrict/importchecker

janwijbrand committed:
copy&paste error

mgedmin commented on setup.py:
```diff
@@ -1,11 +1,15 @@
+test_requirements = ['zope.testrunner']
+
+if sys.version_info.major == 2:
+    test_requirements.append('mock')
```
This sort of dynamic dependency computation used to cause problems with certain versions of pip due to wheel caching.

The modern alternative is to use environment markers: https://hynek.me/articles/conditional-python-dependencies/

mgedmin commented on setup.py:
```diff
@@ -39,8 +43,11 @@ def read(*rnames):
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.3',
```
FWIW other zopefoundation projects are dropping Python 3.3 support.

Also, you're not running the tests on 3.3 (neither from Travis, nor from tox.ini), so I don't think it's right to claim it as a supported version in setup.py.

mgedmin commented on src/importchecker/importchecker.py:
```diff
@@ -17,39 +17,31 @@
-    more_node = node
     name = node.__class__.__name__
-    if name == 'Getattr':
+    childs = ast.iter_child_nodes(node)
```
The usual English spelling is `children` ;)

mgedmin reviewed (approved):
LGTM, with minor comments

janwijbrand committed:
no 3.3 support (it wasn't tested in the first place)

janwijbrand committed:
Add a comment about the `mock` test dep.

It might not be installed when using pip that in some (all?) cases
creates a wheel and caches it from this sdist. Since it is a test
dependecny I think it is ok for now.

janwijbrand commented:
I need to study the environment-markers-way of declaring conditional dependencies more to know how to effectively fix it for importchecker. If it causes problems in real-life I can come back to this and try to fix it.
